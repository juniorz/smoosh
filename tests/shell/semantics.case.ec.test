# ADDTOPOSIX

(exit 3)
echo $? # make sure we're making good ecs
case a in
    ( b ) (exit 4) ;;
    ( * ) ;; # don't alter ec
esac
echo $? # should be 3

(exit 5)
case a$(echo $?>ec) in # observe ec before entering case
    ( b ) (exit 6) ;;
esac
echo $?
[ $(cat ec) = "5" ] || exit 2 # shouldn't have been altered yet!

# make sure the ec is actually visible
false
case a in
    ( a ) echo visible $? ;;
esac

# but make sure that no match cases set the ec to 0
false
case a in
    ( b ) (exit 6) ;;
esac
echo $?

