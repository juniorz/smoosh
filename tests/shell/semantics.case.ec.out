3
0
0
visible 1
0
